### PR TITLE
add ABCModelingEngine

### PR DESCRIPTION
**Pull request by nathanfranklin**
This addresses #47.

- [x] create AbcModelingEngine based on description in wiki
- [x] update use of `name` paramter in methods based on  #48 and #38
- [x] add docstrings

### CODE REVIEW

**Comment by itziakos** [comment]
@nathanfranklin, ping me when this PR is ready for review.

**Comment by nathanfranklin** [comment]
@itziakos ,  this is ready for PR.

I have two concerns:
 - location: is simphony-common/cuds/abc_modeling_engine.py correct?
 - docstrings: how does the doc string sound for the get_XXX (and add_XXX) methods sound

**Comment by itziakos** [comment]
> location: is simphony-common/cuds/abc_modeling_engine.py correct?

I think that this is ok

File: simphony/cuds/abc_modeling_engine.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            given, then all meshes will be iterated over.
+
+        """
+        pass
+
+    @abstractmethod
+    def iter_particle_containers(self, names=None):
+        """ Returns an iterator over a subset or all of the particle containers.
+
+        See get_particle_container for more information.
```
Comment (itziakos) [inline simphony/cuds/abc_modeling_engine.py]: I do not think that his give more info, I would suggest that we remove such comments

**Comment by itziakos** [comment]
minor comments apart from that :+1:

**Comment by nathanfranklin** [comment]
Thanks for the quick response.  Just addressed the comment.

**Pull request merged**